src/basic/terminal-util.c

```diff
@@ -898,11 +898,10 @@ int getttyname_malloc(int fd, char **ret) {
         assert(ret);
 
         r = ttyname_r(fd, path, sizeof path); /* positive error */
-        assert(r >= 0);
         if (r == ERANGE)
                 return -ENAMETOOLONG;
-        if (r > 0)
-                return -r;
+        if (r != 0)
+                return negative_errno();
```

Comment:
Hmm, the man page says `ttyname_r` returns "error number upon error". I don't think we should use errno here.

Reply:
oops